### PR TITLE
Fix fileheader analyzer issues

### LINKED ISSUE #1356: File headers: leading spaces should be ignored in copyright messages on lines other than the first
If the correct header for a file is

    // <copyright file="fish.cs" company="Face">
    //   Copyright (c) Face.
    //   All rights reserved.
    // </copyright>

Then the settings file must contain:

    "copyrightText": "Copyright (c) Face.\n   All rights reserved."

Note that, after the "\n" there must be exactly three spaces to match the indent of the second line of the message.

This was also true in legacy StyleCop, but it would be nicer if you could just have: 

    "copyrightText": "Copyright (c) Face.\nAll rights reserved."

### LINKED ISSUE #1432: SA1633 should allow multi-line comment file headers
The following examples produce no errors in classic, but are flagged by `FileHeaderAnalyzers`.

```csharp
/* <copyright file="Test.cs" company="Test">
 Copyright (c) Test. All rights reserved.
 </copyright> */
```

```csharp
/*
 <copyright file="Test.cs" company="Test">
 Copyright (c) Test. All rights reserved.
 </copyright>
*/
```

### PR DESCRIPTION
**Pull request by vweijsters**
Fixes #1356 
Fixes #1432

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/DocumentationRules/SA1636UnitTests.cs [hunk 1]
```diff
@@ -14,6 +14,32 @@ namespace StyleCop.Analyzers.Test.DocumentationRules
...
+  }
+}
+";
+
+        private const string NoXmlMultiLineHeaderTestSettings = @"
+{
+  ""settings"": {
+    ""documentationRules"": {
+      ""companyName"": ""FooCorp"",
+      ""copyrightText"": ""copyright (c) FooCorp. All rights reserved.\n\nLine #3"",
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/DocumentationRules/SA1636UnitTests.cs]: :bulb: You can replace `FooCorp` with `{companyName}` on this line.

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/DocumentationRules/SA1636UnitTests.cs [hunk 2]
```diff
@@ -14,6 +14,32 @@ namespace StyleCop.Analyzers.Test.DocumentationRules
     /// </summary>
     public class SA1636UnitTests : FileHeaderTestBase
     {
+        private const string MultiLineHeaderTestSettings = @"
+{
+  ""settings"": {
+    ""documentationRules"": {
+      ""companyName"": ""FooCorp"",
+      ""copyrightText"": ""copyright (c) FooCorp. All rights reserved.\n\nLine #3""
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/DocumentationRules/SA1636UnitTests.cs]: :bulb: You can replace `FooCorp` with `{companyName}` on this line.

**Comment by sharwell** [comment]
:question: Can you squash these two commits, since they really combine to form the work here?

**Comment by vweijsters** [comment]
Updated and squashed

**Pull request merged**